BardMusicPlayer.Catalog/CatalogManager.cs

```diff
@@ -0,0 +1,359 @@
+        private bool disposedValue;
+
+        /// <summary>
+        /// Internal constructor; this object is constructed with a factory pattern.
+        /// </summary>
+        /// <param name="dbi"></param>
+        private CatalogManager(LiteDatabase dbi)
+        {
+            this.dbi = dbi;
+            this.disposedValue = false;
+        }
+
+        /// <summary>
+        /// Create a new instance of the CatalogManager based on the given LiteDB database.
+        /// </summary>
+        /// <param name="dbPath"></param>
+        /// <returns></returns>
+        public static CatalogManager CreateInstance(string dbPath)
+        {
+            var dbi = new LiteDatabase(dbPath);
+            MigrateDatabase(dbi);
+
+            return new CatalogManager(dbi);
+        }
+
+        /// <summary>
+        /// This creates a playlist containing songs that match the given tag.
+        /// </summary>
+        /// <param name="tag"></param>
+        /// <returns></returns>
+        public IPlaylist CreatePlaylistFromTag(string tag)
+        {
+            if (tag == null)
+            {
+                throw new ArgumentNullException();
+            }
+
+            var songCol = this.GetSongCollection();
+
+            // TODO: This is brute force and not memory efficient; there has to be a better
+            // way to do this, but my knowledge of LINQ and BsonExpressions isn't there yet.
+            var allSongs = songCol.FindAll();
+            var songList = new List<MMSong>();
+
+            foreach (var entry in allSongs)
+            {
+                if (TagMatches(tag, entry))
+                {
+                    songList.Add(entry);
+                }
+            }
+
+            if (songList.Count == 0)
+            {
+                return null;
+            }
+
+            var dbList = new DBPlaylist()
+            {
+                Name = tag,
+                Songs = songList,
+                Id = null
+            };
+
+            return new DBPlaylistDecorator(dbList);
+        }
+
+        /// <summary>
+        /// This creates a new empty playlist with the given name.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        public IPlaylist CreatePlaylist(string name)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException();
+            }
+
+            var dbList = new DBPlaylist()
+            {
+                Songs = new List<MMSong>(),
+                Name = name,
+                Id = null
+            };
+
+            return new DBPlaylistDecorator(dbList);
+        }
+
+        /// <summary>
+        /// This retrieves a playlist with the given name.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns>The playlist if found or null if no matching playlist exists.</returns>
+        public IPlaylist GetPlaylist(string name)
```

Comment:
same feedback. playlist needs to be kept to the key of playlist. not string. match the input string to a playlist

Reply:
which isn't possible, actually. i can have multiple songs in a playlist

Reply:
if this was meant to be public because the use would know the name of the playlist, that is also bad. the "name of the playlist" is a UUID stored in the DB, with songs routed to the addr of the midis in the DB.